[Wilbur Fong] iP

wilburrito opened the pull request:
## Hello! Duke is here to help you organise your day!

> “Whether you think you can, or you think you can't--you're right.” - Henry Ford

Duke will help free some RAM in your brain by taking note of and tracking tasks that you tell Duke about. It's:

- Easy to use
- **~Fun~** **Exciting** to use
- Text-based

All you have to do is:

1. Download it from [here](https://github.com/wilburrito/ip/releases/tag/v0.1).
2. Open Duke up.
3. Add your tasks according to the specified formats and types of tasks that Duke can track
4. Let it manage the tasks for you!

Best part is, Duke is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders

If you know a little Java, you can use Duke to practice it too. Here's the `main` method:

`public static void main(String[] args) {
        Duke.awaken();
    }`

wilburrito committed:
Level-1

wilburrito committed:
Level-2

wilburrito committed:
Level-3

wilburrito committed:
Level-4

wilburrito committed:
Level-5

wilburrito committed:
Level-6

wilburrito committed:
A-TextUiTesting

wilburrito committed:
Level-7

wilburrito committed:
Level-7

wilburrito committed:
branch-Level-8

wilburrito committed:
Merge branch 'branch-Level-7'

wilburrito committed:
Level-7

wilburrito committed:
Level-8

wilburrito committed:
Level-8

wilburrito committed:
Level-8

wilburrito committed:
Level-8

wilburrito committed:
A-MoreOOP

wilburrito committed:
A-Packages

wilburrito committed:
A-JUnit

wilburrito committed:
A-Jar

wilburrito committed:
A-Jar

wilburrito committed:
branch-A-JavaDoc

wilburrito committed:
branch-A-CodingStandard

wilburrito committed:
branch-Level-9

wilburrito committed:
Merge branch 'branch-A-JavaDoc'

wilburrito committed:
Merge branch 'branch-A-CodingStandard'

wilburrito committed:
Level-9

wilburrito committed:
Merge branch 'branch-Level-9'

wilburrito committed:
Merge branch 'master' of https://github.com/wilburrito/ip

wilburrito committed:
Added Gradle and checkstyle

wilburrito committed:
branch-A-Gradle

wilburrito committed:
A-Gradle

wilburrito committed:
branch-A-Checkstyle

wilburrito committed:
Merge branch 'branch-A-Checkstyle'

wilburrito committed:
Added some JavaFX stuff

wilburrito committed:
branch-Level-10

wilburrito committed:
Merge branch 'branch-Level-10'

kaldius commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,126 @@
+     * loads data about the task list from dukeData.txt.
+     * @return an ArrayList of TaskItems that will be used to create Duke's TaskList.
+     */
+    public static ArrayList<TaskItem> loadData() {
```
Your `loadData()` method is rather long and deeply nested. Abstracting out certain parts to helper methods/classes could make it simpler!

kaldius commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,126 @@
+                    String completed = task.substring(4, 5);
+                    String taskType = task.substring(1, 2);
+
+                    if (taskType.equals("D")) {
```
This section where you check for each type of task and add them to the list could be abstracted away and placed in the `TaskItem` class or your `TaskList` class to reduce bulk in this method.

aishh12 replied:
Yup, I agree with this :)

kaldius commented on src/main/java/duke/TaskItem.java:
```diff
@@ -0,0 +1,64 @@
+     * toFileString returns the TaskItem in a string.
+     * @return returns the Deadline in a string format that is specifically made for the dukeData.txt file.
+     */
+    public String toFileString() {
```
This method seems redundant since you already have a `toString()` method that contains the same lines of code.

kaldius commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,79 @@
+     * @param keyword if a task contains this, it will be printed to the user.
+     * @return the TaskList containing tasks matching the keyword, if any.
+     */
+    public TaskList findTask(String keyword) {
```
Is there any particular reason why you implemented this method this way? It seems to me a simple `for` loop iterating through the ArrayList checking if each task's string representation `.contains()` a certain string would be simpler.

kaldius commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,82 @@
+        String body = "";
+        if (taskListSize == 1) {
+            /* System.out.println("Now you have " + 1 + " task in the list."); */
+            body = "Now you have " + 1 + " task in the list.\n";
```
It might be simpler to just type the entire thing as one string instead of concatenating multiple.

kaldius reviewed (commented):
Hi there! I had a good time reading your code. You have clearly put in a lot of effort and thought into your design. I made a few comments in areas where I have a few suggestions on how you could make your code even better. Overall I think this is a well implemented program! Good work!

aishh12 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,201 @@
+     * @param taskToBeDeleted the index of the task to be deleted.
+     * @throws DukeException An exception stemming from incorrect or awkward input to duke.Duke.
+     */
+    public static String deleteTask(String taskToBeDeleted) throws DukeException {
```
Perhaps you may want to consider defining the deleteTask method in the TaskList class (in order to maintain encapsulation and shortening the code for this class)?

aishh12 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,201 @@
+     * @param keyword keyword used to find matching tasks
+     * @return a TaskList with matching tasks to be printed.
+     */
+    public static TaskList findTask(String keyword) {
```
As mentioned earlier, you may want to consider adding all the methods related to the task list (adding, deleting, marking and finding tasks as well as printing the list) under Tasklist.java instead?

aishh12 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,55 @@
+            } else {
+                timeOfDay = "AM";
+            }
+            this.dateAndTimeInString = this.byDateAndTime.getDayOfWeek().toString() + " " + hour + timeOfDay;
```
Would you consider formatting the date and time to be stored in the "dateAndTimeInString" by using the methods under the DateTimeFormatter class? This way you can make your code cleaner and print in a specified format too :)

aishh12 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,350 @@
+                Ui.customErrorMessage(e.getMessage());
+            }
+        } else {
+            response = "____________________________________________________________\n"
```
Since there are multiple instances to draw the lines for formatting, you could consider having a method to print the line to make your code cleaner!

aishh12 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,82 @@
+    public static String deletedTaskMessage(TaskItem task, int taskListSize) {
+        String body = "";
+        if (taskListSize == 1) {
+            /* System.out.println("Now you have " + 1 + " task in the list."); */
```
Commented code can be removed since it is no longer needed.

aishh12 reviewed (commented):
Overall, good job! Some methods were pretty long and abstracting away parts of it could significantly reduce code length and enhance readability. Besides that it's good to go, all the best :)

wilburrito committed:
Added some assertions in some methods.

Could have been more liberal with the assertions, but this should
suffice for now until I see a greater need to add more assertions.

wilburrito committed:
Merge pull request #1 from wilburrito/branch-A-Assertions

Added some assertions in some methods.

wilburrito committed:
branch-A-CodeQuality

wilburrito committed:
Merge pull request #2 from wilburrito/branch-A-CodeQuality

branch-A-CodeQuality

wilburrito committed:
Fixed code to increase code quality.

Some methods were too long, and needed to be shortened.

Some were inefficient, eg. not using stringbuilder for string
concatenation in a for-loop.

Such changes to the code were effected.

wilburrito committed:
branch-BCD-Extension

wilburrito committed:
Merge branch 'branch-BCD-Extension'

wilburrito committed:
A-BetterGui

wilburrito committed:
A-UserGuide

wilburrito committed:
Image of Ui added to README

wilburrito committed:
Fixed some bugs for Duke

wilburrito committed:
Fixed Ui.png
